splice/queries/reporting.py

```diff
@@ -1,33 +1,88 @@
 from splice.models import impression_stats_daily
-from splice.queries.tile import get_tiles_by_campaign
+from splice.queries.tile import get_tiles
 from sqlalchemy.sql import func
+from sqlalchemy.orm import aliased
+from sqlalchemy import case
 
 
-def get_stats_by_date(campaign_id):
-
+def get_stats(group_by, filters=None):
```

Comment:
Add some doc strings here to outline the function and arguments could be helpful for the API users.